pkg/ingester/shipper.go

```diff
@@ -29,39 +29,29 @@ import (
 	"github.com/grafana/mimir/pkg/storage/tsdb/block"
 )
 
-type metrics struct {
-	dirSyncs                 prometheus.Counter
-	dirSyncFailures          prometheus.Counter
+// shipperMetrics holds the shipper metrics. Mimir runs 1 shipper for each tenant but
+// the metrics instance is shared across all tenants.
+type shipperMetrics struct {
 	uploads                  prometheus.Counter
 	uploadFailures           prometheus.Counter
 	lastSuccessfulUploadTime prometheus.Gauge
 }
 
-func newMetrics(reg prometheus.Registerer) *metrics {
-	var m metrics
-
-	m.dirSyncs = promauto.With(reg).NewCounter(prometheus.CounterOpts{
-		Name: "thanos_shipper_dir_syncs_total",
-		Help: "Total number of dir syncs",
-	})
-	m.dirSyncFailures = promauto.With(reg).NewCounter(prometheus.CounterOpts{
-		Name: "thanos_shipper_dir_sync_failures_total",
-		Help: "Total number of failed dir syncs",
-	})
-	m.uploads = promauto.With(reg).NewCounter(prometheus.CounterOpts{
-		Name: "thanos_shipper_uploads_total",
-		Help: "Total number of uploaded blocks",
-	})
-	m.uploadFailures = promauto.With(reg).NewCounter(prometheus.CounterOpts{
-		Name: "thanos_shipper_upload_failures_total",
-		Help: "Total number of block upload failures",
-	})
-	m.lastSuccessfulUploadTime = promauto.With(reg).NewGauge(prometheus.GaugeOpts{
-		Name: "thanos_shipper_last_successful_upload_time",
-		Help: "Unix timestamp (in seconds) of the last successful TSDB block uploaded to the bucket.",
-	})
-
-	return &m
+func newShipperMetrics(reg prometheus.Registerer) *shipperMetrics {
+	return &shipperMetrics{
+		uploads: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_ingester_shipper_uploads_total",
```

Comment:
Note to reviewers: prefix renamed from `thanos_` to `cortex_` because these metrics are not mapped anymore by `tsdbMetrics` but directly exposed. So, the final exposed metric name hasn't changed.

Reply:
In the changelog you write that `thanos_shipper_dir_syncs_total` is removed, but I don't see such metric and according to this note, there isn't one ?

Reply:
I wrote the wrong metric name in the CHANGELOG 🤦